API/ev3_sensors/utility.c

```diff
@@ -0,0 +1,12 @@
+#include "API/ev3_inputs/ev3_input_uart.h"
+#include "include/ev3_command.h"
+#include "utility.h"
+
+void setUARTSensorHandlerMode(SensorHandler *handler, int port, int type, int mode) {
+    int previousMode = handler->currentSensorMode[port];
+    setUARTSensorMode(port, type, mode);
+    EV3Color->currentSensorMode[port] = mode;
+    if (previousMode != mode) {
+        Wait(200);
```

Comment:
I think (but I may be wrong) that this may not be needed. The UART_DATA_READY flag is set by the kernel once the new data arrives and so after `setUARTSensorMode()` returns, the sensor should be ready.

However, according to http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-buster/sensor_data.html#lego-ev3-us-mode3-value0, it may be useful in preventing sensor lockups if the mode is changed fast enough.